src/components/creator/Editor/SceneEdition/Grid/SceneCell.tsx

```diff
@@ -36,29 +35,30 @@ export const SceneCell: React.FC<CellProps> = (props) => {
 
 
     const handleClick = () => {
-        if (selectedTool === OBSTACLE && cellHasActor){
-            if(!isInitialCell){
-                setCurrentCell(selectedTool)
+        console.log(selectedTool)
+        switch (selectedTool) {
+            case '': break; //by context default
+            case 'O': handleObstacle(); break;
+        }
+    }
+
+    const handleObstacle = () => {
```

Comment:
Me re costó ver por qué es necesario diferenciar los ifs y anidarlos. Podemos agregar un comment " We can't replace actor on the initial cell, so we don't do anything in that case"

o bien ponerlo más directo, así:

```ts
if (cellHasActor && isInitialCell) return; // We can't replace actor on the initial cell
if (cellHasActor && !isInitialCell) relocateActor()
setCurrentContent(selectedTool) // obstacle replaces everything
```

No sé si vale la pena el cambiazo.